src/shared/cluster_utils.c

```diff
@@ -105,3 +104,48 @@ char *get_master_node(void) {
 
     return master_node;
 }
+
+// Get cluster node name from configuration or environment variables
+char * wm_node_name() {
+    const char *(xml_node[]) = {"ossec_config", "cluster", "node_name", NULL};
+    char * node_name;
+
+    OS_XML xml;
+
+    if (OS_ReadXML(DEFAULTCPATH, &xml) < 0){
+        merror_exit(XML_ERROR, DEFAULTCPATH, xml.err, xml.err_line);
+    }
+
+    node_name = OS_GetOneContentforElement(&xml, xml_node);
+    OS_ClearXML(&xml);
+
+    if (!node_name) {
+        return NULL;
+    }
+
+    // Get environment variables
+
+    if (strcmp(node_name, "$NODE_NAME") == 0) {
+        free(node_name);
+        node_name = getenv("NODE_NAME");
+        
+        if (node_name) {
+            return strdup(node_name);
```

Comment:
This `strdup(node_name)` is a duplicate of the string pointed to by `node_name`. It's not necessary, use `return node_name`